src/ui/page_wifi.c

```diff
@@ -455,15 +459,57 @@ static void page_wifi_update_current_page(int which) {
     case 1:
         pp_wifi.p_arr.max = page_wifi.page_2.row_count;
         btn_group_show(&page_wifi.page_2.dhcp.button, true);
+        btn_group_enable(&page_wifi.page_2.dhcp.button, page_wifi.page_1.mode.button.current == WIFI_MODE_STA);
+        if (page_wifi.page_1.mode.button.current != WIFI_MODE_STA) {
+            lv_obj_clear_flag(pp_wifi.p_arr.panel[1], FLAG_SELECTABLE);
+        }
+
         lv_obj_clear_flag(page_wifi.page_2.ip_addr.label, LV_OBJ_FLAG_HIDDEN);
         lv_obj_clear_flag(page_wifi.page_2.ip_addr.input, LV_OBJ_FLAG_HIDDEN);
         lv_obj_clear_flag(page_wifi.page_2.netmask.label, LV_OBJ_FLAG_HIDDEN);
         lv_obj_clear_flag(page_wifi.page_2.netmask.input, LV_OBJ_FLAG_HIDDEN);
         lv_obj_clear_flag(page_wifi.page_2.gateway.label, LV_OBJ_FLAG_HIDDEN);
         lv_obj_clear_flag(page_wifi.page_2.gateway.input, LV_OBJ_FLAG_HIDDEN);
+
+        if (page_wifi.page_1.mode.button.current == WIFI_MODE_AP ||
+                (page_wifi.page_1.mode.button.current == WIFI_MODE_STA &&
+                page_wifi.page_2.dhcp.button.current == 1)) {
+            lv_obj_clear_state(page_wifi.page_2.ip_addr.label, STATE_DISABLED);
+            lv_obj_clear_state(page_wifi.page_2.ip_addr.input, STATE_DISABLED);
+            lv_obj_clear_state(page_wifi.page_2.netmask.label, STATE_DISABLED);
+            lv_obj_clear_state(page_wifi.page_2.netmask.input, STATE_DISABLED);
+            lv_obj_clear_state(page_wifi.page_2.gateway.label, STATE_DISABLED);
+            lv_obj_clear_state(page_wifi.page_2.gateway.input, STATE_DISABLED);
+        } else {
+            lv_obj_add_state(page_wifi.page_2.ip_addr.label, STATE_DISABLED);
```

Comment:
IP Address can be set for DHCP mode. It allows for client to request that specific address to be reserved if DHCP server has not leased it out. Allows for DHCP mode to have a semi-static ip address.

Reply:
Oh, I was unaware that this is a use-case. I just reverted this disabling part.